src/platform/plugins/shared/discover/public/application/main/components/layout/cascaded_documents/blocks/cascade_leaf_component.tsx

```diff
@@ -0,0 +1,338 @@
+
+  return (
+    <div
+      data-test-subj="discoverCascadeCustomDataGridBody"
+      role="rowgroup"
+      css={customCascadeGridBodyStyle.wrapper}
+    >
+      <div css={customCascadeGridBodyStyle.headerRow}>{headerRow}</div>
+      <div
+        ref={customGridBodyScrollContainerRef}
+        css={customCascadeGridBodyStyle.virtualizerContainer}
+      >
+        <div
+          style={{
+            height: virtualizer.getTotalSize(),
+          }}
+        >
+          <div
+            css={customCascadeGridBodyStyle.virtualizerInner}
+            style={{
+              transform: `translateY(${
+                items[0]?.start - (virtualizer.options.scrollMargin ?? 0)
+              }px)`,
+            }}
+          >
+            {items.map((virtualRow) => (
+              <div
+                key={virtualRow.key}
+                data-index={virtualRow.index}
+                ref={virtualizer.measureElement}
+                className="euiDataGridRow"
+                css={customCascadeGridBodyStyle.displayFlex}
+              >
+                {visibleColumns.map((column, colIndex) => {
+                  return (
+                    <Cell
+                      key={`${virtualRow.index}-${colIndex}`}
+                      colIndex={colIndex}
+                      rowIndex={virtualRow.index}
+                      visibleRowIndex={virtualRow.index}
+                      columnId={column.id}
+                    />
+                  );
+                })}
+              </div>
+            ))}
+          </div>
+        </div>
+      </div>
+      <React.Fragment>{footerRow}</React.Fragment>
+    </div>
+  );
+});
+
+export const ESQLDataCascadeLeafCell = React.memo(
+  ({
+    cellData,
+    cellId,
+    dataGridDensityState,
+    showTimeCol,
+    dataView,
+    services,
+    showKeyboardShortcuts,
+    renderDocumentView,
+    externalCustomRenderers,
+    getScrollElement,
+    getScrollMargin,
+    getScrollOffset,
+  }: ESQLDataCascadeLeafCellProps) => {
+    const [sampleSize, setSampleSize] = useState(cellData.length);
+    const [expandedDoc, setExpandedDoc] = useState<DataTableRecord | undefined>();
+    const [cascadeDataGridDensityState, setCascadeDataGridDensityState] = useState<DataGridDensity>(
+      dataGridDensityState ?? DataGridDensity.COMPACT
+    );
+
+    // TODO: Implement column selection logic,
+    // probably requires a new selection component that will be used within the row
+    const [selectedColumns, setSelectedColumns] = useState<string[]>([]);
+
+    const [isCellInFullScreenMode, setIsCellInFullScreenMode] = useState(false);
+
+    const renderCustomToolbarWithElements = useMemo(
+      () =>
+        getRenderCustomToolbarWithElements({
+          leftSide: (
+            <React.Fragment>
+              <EuiText size="s">
+                <b>
+                  <FormattedMessage
+                    id="discover.esql_data_cascade.row.cell.toolbar.heading"
```

Comment:
```suggestion
                    id="discover.cascade.row.cell.toolbar.heading"
```
Looks like we use `discover.cascade` elsewhere for the translation IDs.

Reply:
Settled on camelCase, resolved in 1b592bb9af455d7fe101bddecae08e086431e8cf